Blog post: starter issues

gr2m opened the pull request:
closes https://github.com/hoodiehq/editorial/issues/55

gr2m committed:
feat(blog): Starter issues

janl commented:
This is a killer blog post Gregor! <3

I put it in a Google doc as to quickly annotate a few things: https://docs.google.com/document/d/1Upgo3ys_iTWhR4OuPGXLMu-XlDtfAdiPyNvI7-NYW9s/edit (everyone can add comments here)

But this is very very good, thanks! :)

gr2m commented:
thanks @janl, I incorporated all your suggestions and amended the commit. Can we ship this?

gr2m commented:
I’ve added a section, "Benefits"

> The benefits starter issues go both ways. For people new to the project, it lowers the barrier and makes contributing more fun. For maintainers, it’s a great way to empower more people to do code reviews, as it’s just checking off the list. And creating starter issues might become a part the community can  engage with, which is exactly why we decided to write this blog post :) We hope it will help your Open Source project, too.

I think it’s worth saying that creating this starter issues is not only a "cost" for Open Source maintainers, it comes with the great benefit that it helps to scale up the maintainers team itself. I hope I expressed that correctly in the paragraph? Any thoughts?

gr2m commented:
I made a screencast here: https://www.dropbox.com/s/z9lhk9ajx48e8nj/howto-hoodie-starter-issues.mov
Is it okay like that? Do we have a Vimeo or Youtube account I can upload it to?

espy commented:
Oooh, very nice. We have a Youtube account, if you sign in to youtube with your hoodiefirm address, you should be signed in as the `hoodiehq` user and can simply upload videos to that account. I can help you and do that if it doesn't work for you :)

lewiscowper commented:
I added some suggestions to the google doc, and appended the Benefits section with a tiny change.

janl commented:
+1000 — should we get this published on Tuesday Feb 23 11:00 CET? I’m happy to do the publish & tweets

gr2m commented:
wow thanks for the editing work, I learn so much from this! I’ve updated the post and squashed the commits, this is ready for prime time. Publishing tomorrow, Tuesday Feb 23 11:00 CET sounds great.

Last question: I think we wanted to mention people who helped with editing at the end of each blog post. Can we add it here as well? I think people who helped were @janl @espy @Charlotteis @lewiscowper, hope I didn’t forget anyone

lewiscowper commented:
@gr2m didn't @patriciagarcia have some comments?

gr2m commented:
@kmcrayton7 just gave great input, too: https://github.com/hoodiehq/hood.ie/commit/4809c0ec66072c3c8fe0485c32de1bb58f98a16d

varjmes commented:
@gr2m you're now using "first-timers-only" and "Your First PR" as labels, do you want to mention your reasons for using them in this post?

varjmes commented on _posts/2016-02-18-starter-issues.md:
```diff
@@ -0,0 +1,112 @@
+
+Putting a “starter” label on an issue is Hoodie’s red carpet that we roll out for new and existing contributors. Here is an example “starter” issue: https://github.com/hoodiehq/hoodie-server-account/issues/81. We maintain a list of open starter issues at http://go.hood.ie/hoodie-starter-issues.
+
+It looks like a lot of work compared creating an issue stating the problem, but you get fast at creating these once you get the hang of it. Plus, “starter” issues are often created as follow ups to quick issues created without much description.
```
"faster" instead of "fast" ?

varjmes commented on _posts/2016-02-18-starter-issues.md:
````diff
@@ -0,0 +1,112 @@
+- [ ] update the [README](link-to-readme-section): add `myLib.myMethod` in the same format the other APIs are documented
+```
+
+At Hoodie, we follow [Commit Message Conventions](https://github.com/hoodiehq/hoodie/blob/master/CONTRIBUTING.md#commit-message-conventions). Instead of explaining why and how, we simply make actionable items for the separate commits we’d like to have
````
remove "simply"

varjmes commented on _posts/2016-02-18-starter-issues.md:
````diff
@@ -0,0 +1,112 @@
+- [ ] Make sure that tests fail when run with `npm test`. Commit with `test: myLib.myMethod` and push
+```
+
+Now comes the fun, the “actual” work, and all it takes is an item describing it. If helpful, we link to the right place and explain how the feature / bug fix can be implemented. But it’s okay to keep it short and leave something for the contributor to figure out. It’s fun after all, and at this point the contributor should feel comfortable to ask for help if they get stuck. So often time we simpl ...
````
remove "simply"

varjmes replied:
I'm not sure if we should say "actual" work as this is all really good work from start to finish. "actual" work suggests that all the work they do getting set up and started is not real work, and I feel that this belittles the good work a contributor has done to get this far.

I suggest:

"Now comes the fun, the code a contributor needs to write, and we use a single item on the list to describe this."

gr2m replied:
absolutely true. Thanks so much for catching this! Also all your other comments are great, incorporating all of them right now :)

varjmes commented on _posts/2016-02-18-starter-issues.md:
````diff
@@ -0,0 +1,112 @@
+- [ ] Someone else: review & merge
+```
+
+During the review it often turns out that something is wrong or missing, so the pull request needs amendments. The contributor can simply add more commits as long as needed. Once they are done, they can either squash the commits themselves if they know how to do it, or the reviewer does it for them. Very important when the reviewer is cleaning up the commit messages is to keep the original author ...
````
remove "simply"

varjmes replied:
Instead of:

"During the review it often turns out that something is wrong or missing, so the pull request needs amendments"

I suggest:

"During the review it may turn out something is not quite right, so the pull request needs amendments"

As 1) "wrong" sounds harsh. 2) Suggesting it happens _often_ suggests that our contributors aren't great at what they do (they so are), and we're not great at explaining to them what they have to do (I think we are) :)

varjmes replied:
change " Very important when the reviewer is cleaning up the commit messages is to keep the original author" to "It is very important that when the reviewer is cleaning up the commit messages, that they keep the original author"

varjmes commented on _posts/2016-02-18-starter-issues.md:
````diff
@@ -0,0 +1,112 @@
+- [ ] Commit with `docs(README): myLib.myMethod`
+```
+
+Once the first commit is ready, it’s time to start the Pull Request. Even though we link the tutorial on how to contribute to open source above, we do it again, because it doesn’t hurt. We also congratulate if this is going to be the first Pull Request for someone, because it means a lot to us :)
````
change "We also congratulate" to "We also give congratulations"

varjmes commented on _posts/2016-02-18-starter-issues.md:
````diff
@@ -0,0 +1,112 @@
+If this is your first, welcome :tada: :smile: [Here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request)
+```
+
+Once the PR is created, we check all above boxes and replace the above item with something like `- [x] Start a Pull Request (#123)`, where `#123` is the number of the Pull Request.
````
change "we check all above boxes" to "we check all the above boxes"

varjmes commented on _posts/2016-02-18-starter-issues.md:
````diff
@@ -0,0 +1,112 @@
+- [ ] Add a check that `myLib.myMethod` is a function in [/test/integration/constructor-test.js](https://github.com/myuser/mylib/tree/master/test/integration/constructor-test.js)
+```
+
+Then we ask the user to commit and push the tests. This way the commit will trigger our continuous integration tests on [Travis CI](https://travis-ci.org) and will add a red **×** to the commit, so we can be sure that tests are failing as expected. Here’s [an example](https://github.com/hoodiehq/pouchdb-hoodie-api/pull/100/commits) of what I mean.
````
use contributor instead of "user", maybe?

varjmes commented on _posts/2016-02-18-starter-issues.md:
```diff
@@ -0,0 +1,112 @@
+
+### Benefits
+
+The benefits of starter issues go both ways. For people new to the project, it lowers the barrier and makes contributing more fun. For maintainers, it’s a great way to empower more people to do code reviews, as it’s just checking off the list. More importantly, it is our hope that creating starter issues will become a part of the Open Source culture and a great format for community engagement.
```
remove "just".

I suggest:

"it’s a great way to empower more people to do code reviews, as they have a list they can follow and check items off of"

gr2m replied:
"check items off of" Is there a way to rephrase this without the "off of" at the end? It’s a bit hard to read to me. Maybe we can just leave out the last "of"?

varjmes commented:
@gr2m @janl caught a few language smells like "just" "simply". And a few instances of "let's say this in a slightly nicer way". As always, I hope they are useful.

Thanks for writing this gregor, it seems like you're really enjoying the writing process, something I don't think you felt so confident in doing before, so it's great to see it happen.

janl commented:
+1

gr2m commented:
@Charlotteis 

> @gr2m you're now using "first-timers-only" and "Your First PR" as labels, do you want to mention your reasons for using them in this post?

I only used `first-timers-only` to see how it works with https://twitter.com/first_tmrs_only. I like `@YourFirstPR` much better because it’s not automated, I feel like https://twitter.com/first_tmrs_only has already been abused by some of the issues that lack helpful descriptions. So to answer your question, no I think we don’t need to mention this in the blog post 

> Thanks for writing this gregor, it seems like you're really enjoying the writing process, something I don't think you felt so confident in doing before, so it's great to see it happen.

YES! YES! YES! Loving this <3

varjmes commented:
> I feel like https://twitter.com/first_tmrs_only has already been abused by some of the issues that lack helpful descriptions.

I feel exactly the same way.

gr2m commented:
I’ve added the video (below the "STEP BY STEP" title) and a "Thank you" section to the end of the post. I also tested it locally and adjusted the formatting a bit. I squashed the commits into one.

This is good to go for tomorrow @janl :+1:

janl commented:
We are live: http://hood.ie/blog/starter-issues.html

janl commented:
@gr2m I had to do quite a bit of formatting cleanup, can you try running this through jekyll locally next time? Thanks! <3

gr2m commented:
I did this time, and I fixed a lot of things! What was wrong this time?

janl commented:
@gr2m any code formatting, just ``doesn’t work in our blog, it needs to be`<pre><code>...` for blocks, we don’t have styles for inline code, so I used italics there.
